src/core/core.scale.js

```diff
@@ -1129,9 +1129,12 @@ class Scale extends Element {
 
 			if (isHorizontal) {
 				x = pixel;
-				textOffset = position === 'top'
-					? ((!rotation ? 0.5 : 1) - lineCount) * lineHeight
-					: (!rotation ? 0.5 : 0) * lineHeight;
+				if (position === 'top') {
+					textOffset = -1 * (!rotation ? (lineCount - 0.5) : (lineCount / 2)) * lineHeight;
```

Comment:
This seems weird to me. If there's one degree of rotation we use a completely different formula? I'd expect it to be doing something with `sin` or `cos` of `rotation`

Reply:
Agreed. I only tested this for 0 and 90 degree rotation. Not sure what happens in the 45 degree case. Will investigate

Reply:
Ok, updated this logic. There are two terms: one with `Math.sin()` and one with `Math.cos()`. Works well and I tested at 0, 45, and 90 degrees for top & bottom positions